From 0803374a95be94a09f0b88ac42fbc21bc4af02f3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 7 Aug 2026 03:17:39 +0000
Subject: [PATCH 28/41] opengl (Apple): request a legacy double-buffered pixel
 format pre-10.7

kCGLOGLPVersion_3_2_Core isn't available before macOS 10.7; request a
plain accelerated double-buffered context instead.

Ported from https://github.com/macos-powerpc/powerpc-ports/blob/main/sysutils/fastfetch/files/0005-opengl_apple.c-fix-for-10.7.patch
---
 src/detection/opengl/opengl_apple.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/detection/opengl/opengl_apple.c b/src/detection/opengl/opengl_apple.c
index 0f994175..a0362fcf 100644
--- a/src/detection/opengl/opengl_apple.c
+++ b/src/detection/opengl/opengl_apple.c
@@ -42,7 +42,7 @@ static const char* cglHandlePixelFormat(FFOpenGLResult* result, CGLPixelFormatOb
 const char* cglDetectOpenGL(FFOpenGLResult* result) {
     CGLPixelFormatObj pixelFormat;
     CGLPixelFormatAttribute attrs[] = {
-        kCGLPFAOpenGLProfile, (CGLPixelFormatAttribute) kCGLOGLPVersion_3_2_Core, kCGLPFAAccelerated, 0
+        kCGLPFADoubleBuffer, kCGLPFAAccelerated, 0
     };
 
     GLint num;
-- 
2.43.0

